src/main/java/addon/Task.java

```diff
@@ -0,0 +1,132 @@
+package addon;
+
+import java.time.LocalDate;
+
+/**
+ * Parent class Task, contains doneness and task name
+ */
+public class Task {
+
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Task constructor.
+     * @param description Name of task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String toString() {
+        return ((isDone ? "[X] " : "[ ] ") + this.description);
+    }
+
+    /**
+     * Checks if the description contains the keyword provided.
+     * @param str String to be queried.
+     * @return true if found, false otherwise.
+     */
+    public boolean nameContains(String str) {
+        String[] split = description.split(" ");
+        boolean result = false;
+        for (String s : split) {
+            if (s.equals(str)) {
+                result = true;
+                break;
+            }
+        }
+        return result;
+    }
+
+    public void markDone() {
+        isDone = !isDone;
+    }
+
+    /**
+     * Child class of Task, adds deadline field.
+     */
+    public static class Deadline extends Task {
+
+        protected LocalDate date;
+
+        /**
+         * Deadline constructor.
+         * @param description Name of deadline.
+         * @param by Deadline of task.
+         */
+        public Deadline(String description, LocalDate by) {
+            super(description);
+            this.date = by;
+        }
+
+        @Override
+        public String toString() {
+            return "[D]" + super.toString() + " (by: " + Ui.printDate(date) + ")";
+        }
+
+        public boolean dateEquals(LocalDate theirDate) {
```

Comment:
Should the method be named isDateEquals instead to better describe it as a boolean?